core/station/impl/src/services/system.rs

```diff
@@ -413,8 +411,32 @@ mod install_canister_handlers {
         Ok(())
     }
 
+    pub async fn init_upgrader(
+        input: station_api::SystemUpgraderInput,
+        controllers: Vec<Principal>,
+    ) -> Result<Principal, String> {
+        match input {
+            station_api::SystemUpgraderInput::Id(upgrader_id) => {
+                mgmt::update_settings(mgmt::UpdateSettingsArgument {
+                    canister_id: upgrader_id,
+                    settings: mgmt::CanisterSettings {
+                        controllers: Some(controllers),
+                        ..Default::default()
+                    },
+                })
+                .await
+                .map_err(|e| format!("Failed to set upgrader controller: {:?}", e))?;
```

Comment:
I guess the point of this is to add a fallback controller if a new one is specified, right?